pkg/cmd/upgrade/upgrade_boot.go

```diff
@@ -312,14 +320,69 @@ func (o *UpgradeBootOptions) updateBootConfig(versionStreamURL string, versionSt
 		return errors.Wrapf(err, "failed to fetch upgrade tag %s from %s", upgradeVersion, bootConfigURL)
 	}
 
+	// Set up custom merge driver to ensure that specified files always use the local/dev env version in merges/cherry picks with conflicts
+	err = o.configureGitMergeExcludes()
+	if err != nil {
+		return errors.Wrapf(err, "configuring files (%s) to always use existing version when conflicting with upstream", strings.Join(filesExcludedFromCherryPick, ", "))
+	}
+
 	err = o.cherryPickCommits(configCloneDir, currentSha, upgradeSha)
 	if err != nil {
 		return errors.Wrap(err, "failed to cherry pick upgrade commits")
 	}
-	err = o.excludeFiles(currentSha)
+	return nil
+}
+
+func (o *UpgradeBootOptions) configureGitMergeExcludes() error {
+	gitConfigFile := filepath.Join(o.Dir, ".git", "config")
+	var existingGitConfig string
+	gcExists, err := util.FileExists(gitConfigFile)
```

Comment:
hmm, should that not go through the gitter really using the `git config` command? See also https://github.com/jenkins-x/jx/pull/6572/files#diff-ea63bc4be56a5cd0bf52d8031a36cefcR179

not sure whether it is a good idea to just append this line? Can that never conflict with existing sections?

Reply:
Yeah, I'd be happy to change that to use `git config`. Will do!

Reply:
I'm waiting on #6572 to get in first, since it seems silly to call `git config` directly when you're about to add a helper function for exactly that.